contrib/sarplus/python/tests/test_pyspark_sar.py

```diff
@@ -0,0 +1,390 @@
+import calendar
+import datetime
+import math
+import numpy as np
+import pandas as pd
+import pytest
+import os
+from sklearn.model_selection import train_test_split
+
+from pyspark.sql import SparkSession
+
+from pysarplus import SARPlus, SARModel
+
+def assert_compare(expected_id, expected_score, actual_prediction):
+    assert expected_id == actual_prediction.id
+    assert math.isclose(expected_score, actual_prediction.score, rel_tol=1e-3, abs_tol=1e-3)
+
+@pytest.fixture(scope="module")
+def spark(app_name="Sample", url="local[*]", memory="1G"):
+    """Start Spark if not started
+    Args:
+        app_name (str): sets name of the application
+        url (str): url for spark master
+        memory (str): size of memory for spark driver
+    """
+
+    spark = (
+        SparkSession.builder.appName(app_name)
+        .master(url)
+        .config("spark.jars", os.path.dirname(__file__) + "/../../scala/target/scala-2.11/sarplus_2.11-0.2.6.jar")
+        .config("spark.driver.memory", memory)
+        .config("spark.sql.shuffle.partitions", "1")
+        .config("spark.default.parallelism", "1")
+        .config("spark.sql.crossJoin.enabled", True)
+        .config("spark.ui.enabled", False)
+        # .config("spark.eventLog.enabled", True) # only for local debugging, breaks on build server
+        .getOrCreate()
+    )
+
+    return spark
+
+@pytest.fixture(scope="module")
+def sample_cache(spark):
+    df = spark.read.csv("tests/sample-input.txt", header=True, inferSchema=True)
+
+    path = "tests/sample-output.sar"
+
+    df.coalesce(1)\
+        .write.format("com.microsoft.sarplus")\
+        .mode("overwrite")\
+        .save(path)
+
+    return path
+
+@pytest.fixture(scope="module")
+def header():
+    header = {
+        "col_user": "UserId",
+        "col_item": "MovieId",
+        "col_rating": "Rating",
+        "col_timestamp": "Timestamp",
+    }
+    return header
+
+@pytest.fixture(scope="module")
+def pandas_dummy_dataset(header):
```

Comment:
Maybe reuse some datasets we have for testing? https://github.com/microsoft/recommenders/blob/master/tests/sar_common.py

Reply:
not sure it's worth the dependency?

Reply:
Same as above.